src/megatron/energon/av/av_decoder.py

```diff
@@ -512,37 +539,21 @@ def get_duration(self, get_frame_count: bool = False) -> tuple[float, Optional[i
 
                 video_duration = video_stream.duration
 
-            if input_container.streams.audio:
-                audio_time_base = input_container.streams.audio[0].time_base
-                audio_start_pts = input_container.streams.audio[0].start_time
-                if audio_start_pts is None:
-                    audio_start_pts = 0.0
-
-                audio_duration = input_container.streams.audio[0].duration
-
-            if audio_duration is None and video_duration is None:
+            if video_duration is None:
                 # If duration isn't found in header the whole video is decoded to
                 # determine the duration.
                 packets = [p for p in input_container.demux(video=0) if p.pts is not None]
                 num_frames = len(packets)
                 video_duration = packets[-1].pts + packets[-1].duration
 
-            # Take the largest duration of either video or audio duration
-            if audio_duration is not None and video_duration is not None:
-                duration = max(
-                    int(video_duration - video_start_pts) * video_stream.time_base,
-                    int(audio_duration - audio_start_pts) * audio_time_base,
-                )
-            elif video_duration is not None:
+            if video_duration is not None and video_stream.time_base is not None:
                 duration = int(video_duration - video_start_pts) * video_stream.time_base
-            elif audio_duration is not None:
-                duration = int(audio_duration - audio_start_pts) * audio_time_base
 
             if get_frame_count and num_frames is None:
                 packets = [p for p in input_container.demux(video=0) if p.pts is not None]
                 num_frames = len(packets)
```

Comment:
```suggestion
                num_frames = sum(1 for p in input_container.demux(video=0) if p.pts is not None)
```
saves the list creation of length `num_frames`